app/src/app/manual/combat/backgroundSchema/edit/[schemaId]/page.tsx

```diff
@@ -0,0 +1,220 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import { useRouter } from "next/navigation";
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import ContentBox from "@/layout/ContentBox";
+import Loader from "@/layout/Loader";
+import { api } from "@/app/_trpc/client";
+import { useRequiredUserData } from "@/utils/UserContext";
+import { canChangeContent } from "@/utils/permissions";
+import { UploadButton } from "@/utils/uploadthing";
+import { showMutationToast } from "@/libs/toast";
+import { Input } from "@/components/ui/input";
+import { Button } from "@/components/ui/button";
+import { Label } from "@/components/ui/label";
+import {
+  Form,
+  FormField,
+  FormItem,
+  FormControl,
+  FormMessage,
+} from "@/components/ui/form";
+import { z } from "zod";
+
+export default function EditBackgroundSchemaPage({
+  params,
+}: {
+  params: { schemaId: string };
+}) {
+  const router = useRouter();
+  const schemaId = params.schemaId;
+  const { data: userData } = useRequiredUserData();
+
+  // Check permissions
+  useEffect(() => {
+    if (userData && !canChangeContent(userData.role)) {
+      void router.push("/profile");
+    }
+  }, [userData, router]);
+
+  // Fetch the background schema
+  const { data, isPending, refetch } = api.backgroundSchema.get.useQuery(
+    { id: schemaId },
+    { enabled: !!schemaId && !!userData },
+  );
+
+  if (isPending || !userData || !canChangeContent(userData.role)) {
+    return <Loader explanation="Loading data..." />;
+  }
+
+  return <EditBackgroundSchemaForm schema={data} refetch={refetch} />;
+}
+
+interface EditBackgroundSchemaFormProps {
+  schema: any; // Replace with the actual type of your background schema
+  refetch: () => void;
+}
+
+interface Schema {
+  ocean?: string;
+  ice?: string;
+  dessert?: string;
+  ground?: string;
+  arena?: string;
+  default?: string;
+}
+
+interface BackgroundSchema {
+  schema?: Schema;
+}
+
+const EditBackgroundSchemaForm: React.FC<EditBackgroundSchemaFormProps> = ({
+  schema,
+  refetch,
+}) => {
+  const router = useRouter();
+  const utils = api.useUtils();
+  const [imageUrls, setImageUrls] = useState<Record<string, string>>({
```

Comment:
Maybe also re-use from the validators file, if we add default values of "" to the zod schema? Might not work, not sure.